src/VirtualClient/VirtualClient.Actions/MLPerfTraining/MLPerfTrainingExecutor.cs

```diff
@@ -0,0 +1,515 @@
+        /// </summary>
+        public string DataPath
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(this.DataPath));
+            }
+        }
+
+        /// <summary>
+        /// Number of GPUs to be utilized
+        /// </summary>
+        public string NumberOfGPUs
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(this.NumberOfGPUs));
+            }
+        }
+
+        /// <summary>
+        /// Config file for GPUs
+        /// </summary>
+        public string ConfigFile
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(this.ConfigFile));
+            }
+        }
+
+        /// <summary>
+        /// The MLPerf Training Nvidia code directory.
+        /// </summary>
+        protected string NvidiaDirectoryPath
+        {
+            get
+            {
+                return this.PlatformSpecifics.Combine(this.PlatformSpecifics.PackagesDirectory, "mlperf", "NVIDIA");
+            }
+        }
+
+        /// <summary>
+        /// The MLPerf Training Pytorch code directory.
+        /// </summary>
+        protected string ExecutionDirectoryPath
+        {
+            get
+            {
+                return this.PlatformSpecifics.Combine(this.NvidiaDirectoryPath, "benchmarks", this.Model, "implementations", this.Implementation);
+            }
+        }
+
+        /// <summary>
+        /// The output directory of MLPerf.
+        /// </summary>
+        protected string OutputDirectoryPath
+        {
+            get
+            {
+                return this.PlatformSpecifics.Combine(this.ExecutionDirectoryPath, "results");
+            }
+        }
+
+        /// <summary>
+        /// Initializes the environment for execution of the MLPerf Training workload.
+        /// </summary>
+        protected override async Task InitializeAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            this.Logger.LogTraceMessage($"{this.TypeName}.InitializationStarted", telemetryContext);
+
+            this.ThrowIfPlatformNotSupported();
+
+            await this.ThrowIfUnixDistroNotSupportedAsync(cancellationToken)
+                .ConfigureAwait(false);
+
+            MLPerfState state = await this.stateManager.GetStateAsync<MLPerfState>($"{nameof(MLPerfState)}", cancellationToken)
+                ?? new MLPerfState();
+
+            if (!state.Initialized)
+            {
+                // add user in docker group
+                await this.ExecuteCommandAsync("usermod", $"-aG docker {this.Username}", this.ExecutionDirectoryPath, cancellationToken);
+
+                // Setup Environment
+                await this.SetupEnvironmentAsync(cancellationToken);
+                state.Initialized = true;
+                await this.stateManager.SaveStateAsync<MLPerfState>($"{nameof(MLPerfState)}", state, cancellationToken);
```

Comment:
if you are keeping training and inference separate, would this conflict with the inference state? Unless this is intentional

Reply:
I have changed the state class.